compiler/rustc_parse/src/parser/stmt.rs

```diff
@@ -294,8 +294,10 @@ impl<'a> Parser<'a> {
                         colon_sp,
                         format!(
                             "while parsing the type for {}",
-                            pat.descr()
-                                .map_or_else(|| "the binding".to_string(), |n| format!("`{n}`"))
+                            pat.descr().map_or_else(
+                                || "the binding".into(),
+                                |n| Cow::Owned(format!("`{n}`"))
+                            )
```

Comment:
```suggestion
                            pat.descr().map_or(
                                Cow::Borrowed("the binding"),
                                |n| Cow::Owned(format!("`{n}`"))
                            )
```

Save a closure since the borrowed cow has no overhead